src/hooks/useInitialValues.ts

```
@@ -0,0 +1,93 @@
+import { useEffect, useState } from 'react';
+import { EncounterContext, inferInitialValueFromDefaultFieldValue, isEmpty } from '..';
+import { OHRIFormField, OpenmrsEncounter } from '../api/types';
+import { getHandler } from '../registry/registry';
+import { evaluateAsyncExpression } from '../utils/expression-runner';
+
+export function useInitialValues(
+  formFields: OHRIFormField[],
+  encounter: OpenmrsEncounter,
+  encounterContext: EncounterContext,
+) {
+  const [asyncInitValues, setAsyncInitValues] = useState<Record<string, Promise<any>>>({});
+  const [isFieldEncounterBindingComplete, setIsFieldEncounterBindingComplete] = useState(false);
+  const [initialValues, setInitialValues] = useState({});
+
+  useEffect(() => {
+    const asyncItemsKeys = Object.keys(asyncInitValues);
+    Promise.all(asyncItemsKeys.map(key => asyncInitValues[key])).then(results => {
+      asyncItemsKeys.forEach((key, index) => {
+        if (!isEmpty(results[index])) {
+          initialValues[key] = results[index];
+          const field = formFields.find(field => field.id === key);
+          try {
+            getHandler(field.type).handleFieldSubmission(field, results[index], encounterContext);
```

What happens if `getHandler(field.type)` returns `null` ? I believe you are swallowing the error sending it to the console. I think we should improve this (not limited to this portion of code) in the future to show the error to the dev, something like: `Error processing filed with id: `myFieldId`, Type `myWrongFieldType` has no handler configure or is not supported`.  This kind of things would avoid a lot of errors and spare debugging time

---

I've been thinking about improving error handling with OHRI Forms and I believe it's very imperative to comprehensively support this. We need to configure error boundaries overriding the default boundary defined by the app-shell. I created a ticket for that: https://ohri.atlassian.net/browse/OHRI-1058

---

Thanks